converter/internal/common/validate.go

```diff
@@ -0,0 +1,26 @@
+package common
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/grafana/agent/converter/diag"
+)
+
+func UnsupportedNotDeepEquals(a any, b any, name string) diag.Diagnostics {
```

Comment:
It might be easier for consumers if this took diagnostics as a parameter, similar to `require.Equals` taking `*testing.T` as a parameter: 

```go
func UnsupportedNotDeepEquals(diags *diag.Diagnostics, a any, b any, name string)
```

Reply:
see below discussion about a planned validation discussion and refactor/tune up. https://github.com/grafana/agent/pull/4506#discussion_r1270674998